### PR TITLE
Fix test `02015_async_inserts_2`

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/AsynchronousInsertQueue.h [hunk 1]
```diff
@@ -105,6 +108,7 @@ class AsynchronousInsertQueue : public WithContext
     /// (async_insert_max_data_size setting). If so, then again we dump the data.
 
     std::atomic<bool> shutdown{false};
+    std::atomic<bool> flushes_cancelled{false};
```
Comment (davenger) [inline src/Interpreters/AsynchronousInsertQueue.h:111]: Just a suggestion: "flushes_enabled" or "flushes_disabled" seems to be a better name

**Comment by davenger** [comment]
The test was flaky because flushes could happen based on deadlines and with this fix in the test the flushes will happen only by size threshold. Is my understanding correct?

**Comment by CurtizJ** [comment]
> The test was flaky because flushes could happen based on deadlines and with this fix in the test the flushes will happen only by size threshold. Is my understanding correct?

Almost right. Previously in debug build server could hung for a several seconds and inserts could be flushed in several batches, while test expects them to be flushed in one batch. Now inserts will be flushed also but deadline but with delay added explicitly by `STOP FLUSHES` query.

**Comment by CurtizJ** [comment]
BTW, I forgot that session settings now work for async inserts. Maybe we can avoid adding new query and fix test by correct settings.

**Review by davenger — APPROVED** [review]

**Pull request merged**